src/parser/pegen.ts

```diff
@@ -44,3 +2205,144 @@ export function new_type_comment(p: Parser, c: any): any {
+//     // we don't need to visit it recursively.
+
+//     switch (e->kind) {
+//         case List_kind:
+//             VISIT_CONTAINER(e, List);
+//             return NULL;
+//         case Tuple_kind:
+//             VISIT_CONTAINER(e, Tuple);
+//             return NULL;
+//         case Starred_kind:
+//             if (targets_type == DEL_TARGETS) {
+//                 return e;
+//             }
+//             return _PyPegen_get_invalid_target(e->v.Starred.value, targets_type);
+//         case Compare_kind:
+//             // This is needed, because the `a in b` in `for a in b` gets parsed
+//             // as a comparison, and so we need to search the left side of the comparison
+//             // for invalid targets.
+//             if (targets_type == FOR_TARGETS) {
+//                 cmpop_ty cmpop = (cmpop_ty) asdl_seq_GET(e->v.Compare.ops, 0);
+//                 if (cmpop == In) {
+//                     return _PyPegen_get_invalid_target(e->v.Compare.left, targets_type);
+//                 }
+//                 return NULL;
+//             }
+//             return e;
+//         case Name_kind:
+//         case Subscript_kind:
+//         case Attribute_kind:
+//             return NULL;
+//         default:
+//             return e;
+//     }
+// }
+
+// void *_PyPegen_arguments_parsing_error(Parser *p, expr_ty e) {
+//     int kwarg_unpacking = 0;
+//     for (Py_ssize_t i = 0, l = asdl_seq_LEN(e->v.Call.keywords); i < l; i++) {
+//         keyword_ty keyword = asdl_seq_GET(e->v.Call.keywords, i);
+//         if (!keyword->arg) {
+//             kwarg_unpacking = 1;
+//         }
+//     }
+
+//     const char *msg = NULL;
+//     if (kwarg_unpacking) {
+//         msg = "positional argument follows keyword argument unpacking";
+//     } else {
+//         msg = "positional argument follows keyword argument";
+//     }
+
+//     return RAISE_SYNTAX_ERROR(msg);
+// }
+
+// void *
+// _PyPegen_nonparen_genexp_in_call(Parser *p, expr_ty args)
+// {
+//     /* The rule that calls this function is 'args for_if_clauses'.
+//        For the input f(L, x for x in y), L and x are in args and
+//        the for is parsed as a for_if_clause. We have to check if
+//        len <= 1, so that input like dict((a, b) for a, b in x)
+//        gets successfully parsed and then we pass the last
+//        argument (x in the above example) as the location of the
+//        error */
+//     Py_ssize_t len = asdl_seq_LEN(args->v.Call.args);
+//     if (len <= 1) {
+//         return NULL;
+//     }
+
+//     return RAISE_SYNTAX_ERROR_KNOWN_LOCATION(
+//         (expr_ty) asdl_seq_GET(args->v.Call.args, len - 1),
+//         "Generator expression must be parenthesized"
+//     );
+// }
+
+export function collect_call_seqs(a: expr[], b: AST[], ...attrs: stmtAttrs) {
+    console.log(b, a);
+    if (b === null) {
+        return new Call(new Name("dummy", new Load(), 1, 0, 1, 0), a, [], ...attrs);
```

Comment:
I went down this massive rabbit hole trying to figure out what this `dummy` id was.

Reply:
I started that rabbit hole then decided to park it and climb out early. Once I got to the 4th nested function that seemed to do something with an empty string I bailed 😊